### PR TITLE
[MM-48060] Added context for emoji

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
Small part of the effort to introduce the context logger for the whole app. This PR just adds the context to `app/emoji.go`.

#### Ticket Link
[MM-48060](https://mattermost.atlassian.net/browse/MM-48060)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/emoji.go [hunk 1]
```diff
@@ -37,7 +38,7 @@ const (
 	MaxEmojiOriginalHeight = 1028
 )
 
-func (a *App) CreateEmoji(sessionUserId string, emoji *model.Emoji, multiPartImageData *multipart.Form) (*model.Emoji, *model.AppError) {
+func (a *App) CreateEmoji(c request.CTX, sessionUserId string, emoji *model.Emoji, multiPartImageData *multipart.Form) (*model.Emoji, *model.AppError) {
```
Comment (isacikgoz) [inline app/emoji.go:41]: Could we drop the `sessionUserId`, it's already conveyed in the context right?

**Review by isacikgoz — COMMENTED** [review]
Looks good, just have a question below 👇

File: app/emoji.go (hunk 1 above)
Comment (noxer) [reply app/emoji.go:41]: Yes. In a later step I want to provide additional methods for that. Most (all?) calls that check for permissions are looking for the current session. Providing the context *and* session ID is redundant.
I wanted to focus purely on adding the context for this (and the following) PR(s).

**Review by agnivade — COMMENTED** [review]
This just passes the context but I don't see any logger changes. Are they planned to be done later?

**Comment by noxer** [comment]
@agnivade yes, I wanted to focus on the "hard" part. Changing `mlog` to `c.Logger()` is easy once the context is there. I hoped that splitting it up makes the reviews easier and faster.

**Review by agnivade — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]

**Pull request merged**